src/sisl/geom/nanoribbon.py

```diff
@@ -75,31 +90,49 @@ def nanoribbon(width: int, bond: float, atoms, kind: str = "armchair"):
             ribbon = ribbon.tile(n, 0)
         # Invert y-coordinates
         ribbon.xyz[:, 1] *= -1
+
         # Set lattice vectors strictly orthogonal
         ribbon.cell[:, :] = np.diag(
-            [ribbon.cell[1, 0], -ribbon.cell[0, 1], ribbon.cell[2, 2]]
+            [ribbon.cell[1, 0], -ribbon.cell[0, 1] + vacuum, ribbon.cell[2, 2]]
         )
         # Sort along x, then y
         ribbon = ribbon.sort(axis=(0, 1))
 
+        if kind == "chiral":
+            # continue with the zigzag ribbon as building block
+            n, m = chirality
+            ribbon = ribbon.tile(n + 1, 0)
+            r = ribbon.xyz[1] - ribbon.xyz[width]
+            ribbon.cell[0] += r + (m - 1) * (ribbon.xyz[2] - ribbon.xyz[0])
+            ribbon = ribbon.remove(range(width))
+            # determine rotation angle
+            x = ribbon.cell[0]
+            angle = np.arccos(x.dot([1, 0, 0]) / x.dot(x) ** 0.5)
```

Comment:
isn't this just `x[0] / x.dot(x) ** 0.5`? For clarity?

Reply:
Is that clearer? I'm thinking of the relation $\cos\theta=\frac{a \cdot b}{|a||b|}$. But I'm happy to change...

Reply:
I think if you leave a comment
```
# angle of vectors, x and b=[1, 0, 0]
```
it was just that you don't use the `b` vector in the division, and hence the connection is lost anyways IMHO. :)
Then I think it is pretty clear.